tests/test_scheduler.py

```diff
@@ -1,155 +1,216 @@
+    stub_worker.join()
 
     # slow task should run exactly once, even if we launched 2 schedulers
-    assert len(text) == 1
-
-
-@pytest.mark.skipif(os.getenv("CI") == "true", reason="test skipped on CI")
-@pytest.mark.parametrize("suffix", ["", "_tz"])
-def test_scheduler_daily_time(start_scheduler, start_cli, suffix):
-    start_scheduler("tests.scheduler_configs.daily{}".format(suffix))
+    assert result == 1
+
+    for scheduler in scheduler_list:
+        scheduler.stop()
+
+
+@pytest.mark.parametrize("tz", [True, False])
+def test_scheduler_daily_time(stub_broker, stub_worker, scheduler, scheduler_thread, tz):
+    result = 0
+
+    @remoulade.actor
+    def write_loaded_at():
+        nonlocal result
+        result += 1
+
+    stub_broker.declare_actor(write_loaded_at)
+
+    if tz:
+        scheduler.schedule = [
+            ScheduledJob(
+                actor_name="write_loaded_at",
+                daily_time=(
+                    datetime.datetime.now(pytz.timezone("Europe/Paris")) + datetime.timedelta(seconds=2)
+                ).time(),
+                tz="Europe/Paris",
+            )
+        ]
+    else:
+        scheduler.schedule = [
+            ScheduledJob(
+                actor_name="write_loaded_at",
+                daily_time=(datetime.datetime.utcnow() + datetime.timedelta(seconds=2)).time(),
+            )
+        ]
+    scheduler_thread.start()
 
-    start_cli("tests.scheduler_configs.daily{}".format(suffix), extra_args=["--threads", "1"])
-
-    time.sleep(3)
-
-    fpath = "/tmp/scheduler-daily{}".format(suffix)
+    time.sleep(1)
 
     # should not have run yet
-    with pytest.raises(FileNotFoundError):
-        with open(fpath, "r"):
-            ...
+    assert result == 0
 
     # should run now
-    time.sleep(5)
-
-    with open(fpath, "r") as f:
-        text = [x for x in f.read().split("\n") if x]
+    time.sleep(1)
 
-    assert len(text) == 1
+    assert result == 1
 
     time.sleep(2)
 
     # should not rerun
-    with open(fpath, "r") as f:
-        text = [x for x in f.read().split("\n") if x]
+    assert result == 1
 
-    assert len(text) == 1
 
+def test_scheduler_new_daily_time(stub_broker, stub_worker, scheduler, scheduler_thread):
+    result = 0
 
-@pytest.mark.skipif(os.getenv("CI") == "true", reason="test skipped on CI")
-def test_scheduler_new_daily_time(start_scheduler, start_cli):
-    start_scheduler("tests.scheduler_configs.daily")
+    @remoulade.actor
+    def write_loaded_at():
+        nonlocal result
+        result += 1
 
-    start_cli("tests.scheduler_configs.daily_new", extra_args=["--threads", "1"])
+    stub_broker.declare_actor(write_loaded_at)
 
-    time.sleep(10)
+    scheduler.schedule = [
+        ScheduledJob(
+            actor_name="write_loaded_at",
+            daily_time=(datetime.datetime.utcnow() - datetime.timedelta(seconds=1)).time(),
+        )
+    ]
+    scheduler_thread.start()
+    time.sleep(1)
```

Comment:
are all these time sleep really usefull ? because they can really slow down the test suite

Reply:
I will try to minimize every delay